app/src/Liquidator.ts

```diff
@@ -275,17 +269,38 @@ export default class Liquidator {
    * @param borrowers The borrowers to scan.
    */
   private async scanBorrowers(): Promise<void> {
-    for (const borrower of this.borrowers) {
-      this.limiter.schedule(async () => {
-        const healthResult = await this.calculateBorrowerHealth(borrower);
-        winston.log(
-          "debug",
-          `#${this.uniqueId} Scan: ${healthResult.borrower} is ${healthResult.isHealthy ? "healthy" : "unhealthy"} (${healthResult.health})`
-        );
-        if (!healthResult.isHealthy) {
-          this.liquidateBorrower(healthResult.borrower);
+    const contractCallContext: ContractCallContext[] = this.borrowers.map((borrower) => {
+      return {
+        reference: borrower,
+        contractAddress: MARGIN_ACCOUNT_LENS_ADDRESS,
+        abi: MarginAccountLensABIJson,
+        calls: [
+          {
+            methodName: "getHealth",
+            methodParameters: [borrower, true],
+            reference: borrower,
+          }
+        ]
+      };
+    });
+    // Get the health of each borrower
+    const results = (await this.multicall.call(contractCallContext)).results;
+    // Check the health of each borrower and liquidate them if they're insolvent
+    for (const result of Object.entries(results)) {
+      const borrower = result[0];
+      const healthResults = result[1].callsReturnContext[0].returnValues;
```

Comment:
What's the data type here? Are you sure the `hexToNumberString` stuff is necessary?

Reply:
The data type is an array of deepClones of BigNumbers that are represented as hex. Unfortunately Big cannot take a hex value directly meaning we first have to convert it to a string.